docs/sdk-configuration.md

```diff
@@ -0,0 +1,492 @@
+      .addSpanProcessor(CustomAttributeAddingProcessor.create())
+      .addSpanProcessor(CustomEventAddingProcessor.create())
+      .setTraceSampler(config.getSamplingRate() != 0 
+          ? Sampler.rateLimiting(config.getSamplingRate()) : Sampler.getDefault())
+      .setTraceLimits(TraceLimits.builder().setMaxAttributes(config.getMaxSpanAttributes()).build())
+      .setIdGenerator(TimestampedIdGenerator.create())
+      .setMeterStuff(InProgressMetricsStuff.create())
+      .build();
+  }
+  
+  @Bean
+  public TracerProvider tracerProvider(OpenTelemetry openTelemetry) {
+    return openTelemetry.getTracerProvider();
+  }
+  
+  @Bean
+  public MeterProvider tracerProvider(OpenTelemetry openTelemetry) {
+    return openTelemetry.getMeterProvider();
+  }
+  
+  @Bean
+  public AuthServiceStub authService(OpenTelemetry openTelemetry, AuthConfig config) {
+    return AuthServiceGrpc.newBlockingStub(ManagedChannelBuilder.forEndpoint(config.getEndpoint()))
+      .withInterceptor(TracingClientInterceptor.create(openTelemetry));
+  }
+  
+  @Bean
+  public ServletFilter servletFilter(OpenTelemetry openTelemetry) {
+    return TracingServletFilter.create(openTelemetry);
+  }
+}
+
+// Use some instrumented client
+@Component
+public class MyAuthInterceptor {
+
+  private final AuthServiceStub authService;
+  
+  @Inject
+  public MyAuthInterceptor(AuthServiceStub authService) {
+    this.authService = authService;
+  }
+  
+  public void doAuth() {
+    if (!authService.getToken("credential").isAuthenticated()) {
+      throw new HackerException();
+    }
+  }
+}
+
+// Use tracer directly, not so common
+@Component
+public class MyService {
+
+  private final Tracer tracer;
+  private final Meter meter;
+  
+  @Inject
+  public MyService(TracerProvider tracerProvider, MeterProvider meterProvider) {
+    tracer = tracerProvider.get("my-service");
+    meter = meterProvider.get("my-service");
+  }
+  
+  public void doLogic() {
+    Span span = tracer.spanBuilder("logic").startSpan();
+    try (Scope ignored = span.makeCurrent()) {
+      Thread.sleep(1000);
+    } finally {
+      span.end();
+    }
+  }
+}
+```
+
+## The global instance of the SDK
+
+A built instance is convenient to use in most Java apps because of dependency injection. Because it
+has a easy-to-reason lifecycle, being tied into the dependency ordering (even if dependency 
+injection happened to be done manually through constructor invocation), we encourage application
+developers to only use it.
+
+However, there are corner cases where an instance cannot be injected. The famous example is MySQL -
+MySQL interceptors are initialized by calling a default constructor, and there is no way to pass a
+built instance of a signal provider. For this case, we must store an SDK instance into a global
+variable. To reduce gotchas for application developers, the first SDK that is instantiated is stored
+automatically in the global variable.
```

Comment:
I think this is a very unexpected behavior....

Reply:
It could also be reasonable to leave this up to frameworks - but it will almost always be required.

Reply:
I agree with Bogdan. Let it be explicit.

Reply:
By "Let it be explicit" you mean "require the SDK user to explicitly assign the global"? Does this mean that the operator/app developer will need to understand the internals of all the instrumentation that happens to be included with the project?

I'm not necessarily arguing in favor of auto-setting the global instance, but I wanted to bring up a problem with the alternative.

My opinion would probably be: get rid of the global altogether and make instrumentation authors work around it in some way [I'm definitely not an expert here...just trying to optimize for simplicity as much as possible].